### PR TITLE
Add deprecated syntax warning type

### PR DESCRIPTION
**Pull request by sovdeeth**
### Description

For use when we intend to remove a syntax in a future version. The warning is locally, but not globally, suppressible for those who refuse to update. 

I chose to make it a static method, as opposed to an automatic warning in the parser, so that we can call it with whatever message is appropriate in a element's init method.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

**Review by Moderocky — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Review by AyhamAl-Ali — APPROVED** [review]
Nice work. I will support this PR later by implementing the required annotations for docs

**Review by UnderscoreTud — APPROVED** [review]
Can you add this to ExprVectorArithmetic while you're at it

**Comment by sovdeeth** [comment]
> Can you add this to ExprVectorArithmetic while you're at it

sure

File: src/main/java/org/skriptlang/skript/lang/script/ScriptWarning.java [hunk 1]
```diff
@@ -18,15 +18,49 @@
...
+	 * Intended for use in {@link ch.njol.skript.lang.SyntaxElement#init(Expression[], int, Kleenean, SkriptParser.ParseResult)}.
+	 *
+	 * @param message the warning message to print.
+	 */
+	public static void printDeprecationWarning(String message) {
+		ParserInstance parser = ParserInstance.get();
+		Script currentScript = parser.isActive() ? parser.getCurrentScript() : null;
+		if (currentScript != null && currentScript.suppressesWarning(ScriptWarning.DEPRECATED_SYNTAX))
+			return;
+		Skript.warning(message);
```
Comment (AyhamAl-Ali) [inline src/main/java/org/skriptlang/skript/lang/script/ScriptWarning.java:63]: Can we add a spceial prefix? like `[Deprecated]`

Comment (sovdeeth) [reply src/main/java/org/skriptlang/skript/lang/script/ScriptWarning.java:63]: Good idea, should help people realize they can suppress it

**Pull request merged**